pennylane/ops/qubit/matrix_ops.py

```diff
@@ -72,6 +72,7 @@ def __init__(self, *params, wires, do_queue=True):
             if not qml.math.allclose(
                 qml.math.dot(U, qml.math.T(qml.math.conj(U))),
                 qml.math.eye(qml.math.shape(U)[0]),
+                atol=1e-6,
```

Comment:
when using 32-bit precision, truly unitary matrices where still raising warnings.

Reply:
Worth leaving this as a comment here, personally I'd be curious if I saw `1e-6` as the tolerance being used.

Reply:
@albi3ro ?